Update release notes for BC4

craigtaverner opened the pull request:
Regenerated release notes for BC4, and *ten* new PRs were added. See generated docs at https://elasticsearch_87635.docs-preview.app.elstc.co/diff

craigtaverner committed:
Update release notes for BC4

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

craigtaverner committed:
Adding missing changelog for geo_grid

craigtaverner committed:
Added missing highlight notes for 84250

kilfoyle commented on docs/reference/release-notes/highlights.asciidoc:
```diff
@@ -29,6 +29,42 @@ for compliance or regulatory reasons, the occasional lookback or investigation,
+[[new_geo_grid_query]]
+=== New geo_grid query
+With the geo grid query, you can now natively return all the documents that overlap a specific geo tile.
+There is no need to reconstruct the geometry or the actual boundaries of the spatial cluster as Elasticsearch
+can do this for you which saves you time and reduces complexity.
+This is especially useful when geometries are spread across tiles like on a soccer ball football.
+While hexagon tiles line the sphere each tile is not exactly straight forward to calculate its boundary.
```
```suggestion
There is no need to reconstruct the geometry or the actual boundaries of the spatial cluster as Elasticsearch
can do this for you, which saves you time and reduces complexity.
This is especially useful when geometries are spread across tiles like on a soccer ball or football.
While hexagon tiles line the sphere, calculating the boundary of each tile is not straightforward.
```

craigtaverner replied:
The text came from PM from a planned blog, but I think your improvements are great, so I just added them. We'll see if PM want to improve the blog later.

kilfoyle commented on docs/reference/release-notes/highlights.asciidoc:
```diff
@@ -29,6 +29,42 @@ for compliance or regulatory reasons, the occasional lookback or investigation,
+=== Ingest performance improvement
+We have improved the pipeline execution logic for pipelines with processors that are synchronous
+by avoiding (deep) recursion.
+On our nightly benchmark that simulate a Logging use-case, this resulted in a 10% reduction of
```
```suggestion
On our nightly benchmark that simulates a Logging use-case, this resulted in a 10% reduction of
```

kilfoyle reviewed (approved):
LGTM! 
Though I'd suggest waiting for a confirmation from @debadair since I'm not yet familiar with the Elasticsearch release notes process. :)

craigtaverner committed:
Update docs/reference/release-notes/highlights.asciidoc

Co-authored-by: David Kilfoyle <41695641+kilfoyle@users.noreply.github.com>

craigtaverner committed:
Update docs/reference/release-notes/highlights.asciidoc

Co-authored-by: David Kilfoyle <41695641+kilfoyle@users.noreply.github.com>

craigtaverner committed:
Backported fixes to original yaml

craigtaverner committed:
Control sort order of release highlights

So that small changes to an individual highlight don't completely shuffle the entire document.

We also added links to the PRs from the highlight titles, for convenience.
Otherwise readers need to search the release notes for the changelog entry and click there,
which is a lot more work.

craigtaverner committed:
Fixed failing test after new ordered highlights

Also made test verify re-ordering of highlights
